### DISCUSSION TITLE
2.X 版本CPU指令集限制说明

### DISCUSSION THREAD

**Discussion by linrunqi08** [alibaba/ilogtail#1398] [Show and tell]
最近有收到一些社区用户反馈运行2.0版本的时候会遇到Illegal instruction的报错，我们非常重视这个问题，但是苦于不能复现，因此希望遇到这个问题的用户可以帮助我们去准确定位和解决这个问题，目前怀疑是跟CPU的指令集有关，希望有遇到这个问题的用户可以执行下lscpu，然后把信息提供给我们，当然如果有我们可以访问的环境会更好，或者我们可以拉一个视频会议一起看下环境的信息

**Comment by linrunqi08** [comment]
已知问题有：
https://github.com/alibaba/ilogtail/discussions/1389
https://github.com/alibaba/ilogtail/issues/1370

**Reply by EEElevation** [reply]
![image](https://github.com/alibaba/ilogtail/assets/52227492/7b88d892-c744-4264-b6a9-532ca1233bea)

![image](https://github.com/alibaba/ilogtail/assets/52227492/a49da4ce-6c09-45ba-ad2c-7516fa6089b4)

**Reply by linrunqi08** [reply]
@EEElevation  非常感谢反馈，你这两个环境都是aarch64的环境，我们现在2.0版本aarch64支持的cpu最低是ARMv8.2-A架构的，因此你下面的环境是可以运行的，上面的环境不可以运行。如果你需要在上面的环境上运行，建议你使用1.8版本

**Comment by linrunqi08** [comment]
2.0.0版本，x86环境下，需要CPU支持avx2指令集; arm环境下，CPU最低需要是ARMv8.2-A架构
2.0.X版本中，x86环境下，需要CPU支持sse4_2指令集; arm环境下，CPU最低需要是ARMv8.2-A架构

不符合以上标准的环境，可以使用1.8.x版本。

**Comment by XuHandsome** [comment]
<img width="1522" alt="image" src="https://github.com/user-attachments/assets/ddecb896-866d-46c6-ba1e-76df375e11a8">

**Reply by XuHandsome** [reply]
我这儿daemonset启动方式，容器内执行`lscpu`有`sse4_2`为啥也报错
/usr/local/ilogtail/ilogtail_control.sh: line 86:    11 Illegal instruction

用的镜像是`sls-opensource-registry.cn-shanghai.cr.aliyuncs.com/ilogtail-community-edition/ilogtail:2.0.4`

**Reply by XuHandsome** [reply]
<img width="746" alt="image" src="https://github.com/user-attachments/assets/1fc8b546-e9ce-44fd-bbea-55ffc1c612cc">

**Comment by weinshun** [comment]
在我的设备中，也有这个报错： Illegal instruction (core dumped)
lscpu有sse4_2为啥也报错，

版本：release的ilogtail-2.0.0和ilogtail-2.0.7 

![image](https://github.com/user-attachments/assets/8396ab20-7bd3-4538-b7a6-16fe2733b530)

![image](https://github.com/user-attachments/assets/d32a848f-2cbb-4f8e-ad2f-7e45a9f7afab)

如需要更多环境信息可与我联系，可以提供远程，可会议。

**Reply by linrunqi08** [reply]
帮忙这样执行一下命令，首先安装gdb，然后进行如下操作
1. gdb ./ilogtail 进入gdb里面
2. 执行run命令
3. 挂了之后，输入disassemble，然后截图看下 @weinshun

**Reply by weinshun** [reply]
好的，这是执行的截图。
![image](https://github.com/user-attachments/assets/ff40e9e2-191a-4799-9ae2-db87ed1fd030)

**Comment by linrunqi08** [comment]
2.0.X版本中，x86环境下，需要CPU支持sse4_2,avx指令集; arm环境下，CPU最低需要是ARMv8.2-A架构

**Reply by xingshao-ye** [reply]
2.0.x版本，x86环境，是要CPU都支持sse4_2,avx指令集吗？
<img width="685" alt="image" src="https://github.com/user-attachments/assets/b5ed4a18-aded-4e47-ae54-fff0502687bf" />
![image](https://github.com/user-attachments/assets/9ac0c8ad-abd4-4038-9ffb-ed26ad60bc5e)
